nixos/modules/services/networking/ncps.nix

```diff
@@ -256,10 +381,10 @@ in
         UMask = "0066";
       };
       script =
-        (lib.optionalString (cfg.cache.dataPath != "/var/lib/ncps") ''
-          if ! test -d ${cfg.cache.dataPath}; then
-            mkdir -p ${cfg.cache.dataPath}
-            chown ncps:ncps ${cfg.cache.dataPath}
+        (lib.optionalString (cfg.cache.storage.s3 == null && cfg.cache.storage.local != "/var/lib/ncps") ''
+          if ! test -d ${cfg.cache.storage.local}; then
+            mkdir -p ${cfg.cache.storage.local}
+            chown ncps:ncps ${cfg.cache.storage.local}
           fi
         '')
```

Comment:
You may consider using `systemd.tmpfiles.settings` to in order to optionally create this directory in a more declarative fashion.